### PR TITLE
Remove copy_version_files.sh

### PR DESCRIPTION
**Pull request by omajid**
Arcade provides this via the GenerateNativeVersionFile target, which we already call in various places. Lets remove this file.

This file, if used, will also cause issues in a VMR build without a git repo. In that mode, git commands do not work, but Arcade and Source Link will still provide git commit and related information correctly.

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
This script is meant for bootstrapping scenarios where we don't have a .NET runtime available for the platform and for innerloop development where build-runtime.sh is invoked directly (without going through the root build script).

Can we instead condition it to safely fail if it's in a VMR-like scenario?

**Comment by omajid** [comment]
Thanks for that context! It doesn't seem to get invoked at all during a normal (aka top-level `./build.sh`) or VMR build. I thought it was some leftover code. I guess I can just close this PR. Or maybe add a comment with that context to the script?

**Comment by am11** [comment]
> Can we instead condition it to safely fail if it's in a VMR-like scenario?

or just:
```sh
if command -v git > /dev/null; then
  call_version_script
done
```

> and for innerloop development where build-runtime.sh is invoked directly

I think we can dedup it now, since `src/coreclr/build-runtime.sh` sources `eng/native/build-commons.sh`.

**Comment by omajid** [comment]
> ```
> if command -v git > /dev/null; then
> ```

I would prefer to avoid this. It's possible to attempt to build the git-archive of the VMR, while the `git` tool is installed. If we want to avoid invoking this script, that would do the wrong thing.

**Comment by am11** [comment]
`if git status > /dev/null 2>&1; then` updated

**Comment by jkoritzinsky** [comment]
> `if git status > /dev/null 2>&1; then` updated

I think @am11's idea here is good as it will also work in the Microsoft VMR build environment if that is ever necessary.

**Comment by omajid** [comment]
I am not sure about that either.

Chances are, devs are working in a git repo - if it isn't the VMR, it contains the scripts to create the VMR archive and/or build the VMR. So `git status` might report version of an unrelated repository.

I think it would be better to make sure that this script is never called in a full/outloop/VMR scenario at all.

**Comment by am11** [comment]
I think it is fine not to support every perceivable scenario. We have only handful of runtime build scenarios which are supported and frequently tested.

For VMR, another approach is add a fallback, something like `DOTNET_RUNTIME_COMMIT` here: https://github.com/dotnet/runtime/blob/8b04e76468af5905f739506e97039170b7554671/eng/native/version/copy_version_files.sh#L12 set `DOTNET_RUNTIME_COMMIT` in VMR (there, we likely have that context in msbuild scripts which runtime commit is in use)  and leave the call-sites intact (unconditioned).

**Comment by am11** [comment]
set: https://github.com/dotnet/dotnet/compare/main...am11:dotnet:patch-1
get: https://github.com/dotnet/runtime/compare/main...am11:runtime:patch-29

Something like that.

**Comment by am11** [comment]
I see. BTW, that `runtimeGitCommitHash` is what sync bot updates, e.g. https://github.com/dotnet/dotnet/commit/3efd8ca08038f193bc108a13aca0c0e9603ac3a0 and since stuff like `<EnvironmentVariables Include="MSBUILDDISABLENODEREUSE=1" />` is likely to stay in VMR, so can `RUNTIME_GIT_COMMIT`?

Note that this script also updates [SCCSID](https://en.wikipedia.org/wiki/Source_Code_Control_System) in runtime and diagnostics repos and altering this will require retesting those scenarios end-to-end. In addition to bootstrapping, libSOS in diagnostics actually depends on the value of that `sccsid[]` in a way that it breaks the debugging functionality, so it is pretty vital. That's why we want to keep this non-msbuild version evaluation working.

**Comment by omajid** [comment]
Backing up a bit:

- If I close this PR without merging, the current behaviour stays.
- That is: we always run this copy_version_files script.
  - If this is an outerloop/VMR build, arcade has already generated the version file, like it always does. It already generates it correctly. This script can already generate or not generate the correct version file, call git or not, and nothing really happens, because the arcade-generated version.c file is used.
    - Arcade (with sourcelink) already handles VMR git vs archive versions/commit hashes correctly.
    - There's no need to change anything.
  - If this is an an innerloop/bootstrap build, then this script already does the right thing. There's no need to change anything

In other words, it seems like there's nothing to do here and everything is already okay?

**Comment by huoyaoyuan** [comment]
Note: this file is currently annoying for WSL and cross compiling Android on Windows. I'd like to see it become more universally available.

**Comment by am11** [comment]
@huoyaoyuan, this script is only meaningful in the context of official product build (the one with `OfficialBuildId` etc. set), so unless it is breaking the build, ignore it.

@omajid, that sounds even better. Lets keep it as-is if it's not causing any issues.
